src/pyodide.js

```diff
@@ -291,7 +305,37 @@ globalThis.languagePluginLoader = (async () => {
     if (!Array.isArray(names)) {
       names = [ names ];
     }
+    // get shared library packages and load those first
+    // otherwise bad things happen with linking them in firefox.
+    sharedLibraryNames = [];
+    try {
+      sharedLibraryPackagesToLoad =
+          recursiveDependencies(names, messageCallback, errorCallback, true);
+      for (pkg of sharedLibraryPackagesToLoad) {
+        sharedLibraryNames.push(pkg[0]);
+      }
+    } catch (e) {
+      // do nothing - let the main load throw any errors
+    }
+    let allLibs = [];
+    let oldOpen = Module.FS.open;
+    newOpen = (path, flags, mode) => {
+      allLibs.push(path);
+      return oldOpen(path, flags, mode);
+    };
+    Module.FS.open = newOpen;
     let promise = loadPackageChain.then(
+        () => _loadPackage(sharedLibraryNames, messageCallback || console.log,
+                           errorCallback || console.error));
+    loadPackageChain = loadPackageChain.then(() => promise.catch(() => {}));
+    await promise;
+    Module.FS.open = oldOpen;
+    for (let newLib of allLibs) {
+      await Module.loadDynamicLibrary(
+          newLib,
+          {global : true, nodelete : true, fs : Module.FS, async : true});
+    }
+    promise = loadPackageChain.then(
```

Comment:
We are calling `_loadPackage` twice here. Is this intentional?

Reply:
Yes, first time to load dynamic library modules, second time to load any other modules. It actually loads the dynamic library modules twice right now, I guess they could be ignored for the second load though.

Reply:
Ahh, yes, sure, the argument is different, I missed that. Sounds good.